io-engine/src/core/logical_volume.rs

```diff
@@ -1,34 +1,54 @@
-///  LogicalVolume Trait Provide all the Generic Interface for Volume
-pub trait LogicalVolume {
-    /// Returns the name of the Logical Volume
+use crate::{core::Protocol, pool_backend::PoolBackend};
+
+/// LogicalVolume Trait Provide all the Generic Interface for a Logical Volume
+/// on any backend type.
+pub trait LogicalVolume: std::fmt::Debug {
+    /// Returns the name of the Logical Volume.
     fn name(&self) -> String;
 
-    /// Returns the UUID of the Logical Volume
+    /// Returns the UUID of the Logical Volume.
     fn uuid(&self) -> String;
 
-    /// Returns the pool name of the Logical Volume
+    /// Returns the pool name of the Logical Volume.
     fn pool_name(&self) -> String;
 
-    /// Returns the pool uuid of the Logical Volume
+    /// Returns the pool uuid of the Logical Volume.
     fn pool_uuid(&self) -> String;
 
     /// Returns entity id of the Logical Volume.
     fn entity_id(&self) -> Option<String>;
 
-    /// Returns a boolean indicating if the Logical Volume is thin provisioned
+    /// Returns a boolean indicating if the Logical Volume is thin provisioned.
     fn is_thin(&self) -> bool;
 
-    /// Returns a boolean indicating if the Logical Volume is read-only
+    /// Returns a boolean indicating if the Logical Volume is read-only.
     fn is_read_only(&self) -> bool;
 
-    /// Return the size of the Logical Volume in bytes
+    /// Return the size of the Logical Volume in bytes.
     fn size(&self) -> u64;
 
     /// Return the committed size of the Logical Volume in bytes.
     fn committed(&self) -> u64;
 
-    /// Returns Lvol disk space usage
+    /// Returns Lvol disk space usage.
     fn usage(&self) -> LvolSpaceUsage;
+    /// Returns the backend type which owns this Logical Volume.
+    fn backend(&self) -> PoolBackend;
+
+    /// Returns a boolean indication if the Logical Volume is a snapshot.
+    fn is_snapshot(&self) -> bool;
+    /// Returns a boolean indication if the Logical Volume is a clone.
+    fn is_clone(&self) -> bool;
+
+    /// Returns its parent snapshot uuid if the Logical Volume is a clone.
+    fn snapshot_uuid(&self) -> Option<String>;
+
+    /// Return the share protocol of this Logical Volume.
+    fn share_proto(&self) -> Protocol;
```

Comment:
`share_proto` --> `share_protocol` proto is a different entity even though it might not be related to this.